pkg/operators/kubenameresolver/kubenameresolver.go

```diff
@@ -134,6 +138,150 @@ func (m *KubeNameResolverInstance) EnrichEvent(ev any) error {
 	return nil
 }
 
+func (k *KubeNameResolver) GlobalParams() api.Params {
+	return nil
+}
+
+func (k *KubeNameResolver) InstanceParams() api.Params {
+	return nil
+}
+
+func (k *KubeNameResolver) Priority() int {
+	return priority
+}
+
+type k8sAccesors struct {
+	PodName   datasource.FieldAccessor
+	Namespace datasource.FieldAccessor
+	PodIP     datasource.FieldAccessor
+	HostIP    datasource.FieldAccessor
+	PodLabels datasource.FieldAccessor
+}
+
+func (k *KubeNameResolver) InstantiateDataOperator(gadgetCtx operators.GadgetContext, instanceParamValues api.ParamValues) (operators.DataOperatorInstance, error) {
+	logger := gadgetCtx.Logger()
+	accessors := make(map[datasource.DataSource]k8sAccesors)
+	for _, ds := range gadgetCtx.GetDataSources() {
+		logger.Debugf("KubeNameResolverOperator inspecting datasource %q", ds.Name())
+
+		if enable, ok := ds.Annotations()["kubenameresolver.enable"]; !ok || enable != "true" {
+			logger.Debugf("KubeNameResolverOperator not enabled by annotation")
+			continue
+		}
+
+		k8sField := ds.GetField("k8s")
+		if k8sField == nil {
+			logger.Debugf("> no k8s fields found")
+			continue
+		}
+
+		k8sAccesors := k8sAccesors{}
+		var err error
+
+		k8sAccesors.PodName = ds.GetField("k8s.podName")
+		if k8sAccesors.PodName == nil {
+			logger.Warnf("No podName field found in datasource %q", ds.Name())
```

Comment:
Do we need to print all these messages as warnings? Is it really and issue for the datasource not to have this field? I mean, is this information important for the user?

Reply:
I am not sure. This was suggested in https://github.com/inspektor-gadget/inspektor-gadget/pull/4169#discussion_r1981147925 but I see both arguments and I can't decide between the two

Reply:
I don't have a strong opinion as well, so let's leave them.